modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyHandler.java

```diff
@@ -213,14 +255,21 @@ public SnapshotPartitionsVerifyHandler(GridCacheSharedContext<?, ?> cctx) {
                         // There is no `primary` partitions for snapshot.
                         PartitionKeyV2 key = new PartitionKeyV2(grpId, partId, grpName);
 
+                        GridIterator<CacheDataRow> partRowIter;
+                        if (punchHoleEnabled && meta.isGroupWithCompresion(grpId) && type() == SnapshotHandlerType.CREATE)
+                            partRowIter = snpMgr.partitionRowIterator(snpCtx, grpName, partId, pageStore, true);
```

Comment:
This makes values hash calculation unskippable. If we don't want hash calculation we can use lightweight check `checkPartitionsPageCrcSum`. I propose to change logic in this method a little bit:
- undo changes in partitionRowIterator, this iterator should only be called on restore operation (skipHash == false)
- add `checkPartitionsPageCrcSum` with hole punching for create operation (at least when group is compessed and punch holes is enabled, but perhaps always, I don't know why we do CRC check for index partition and skip this check for regular partition)
- undo dedicated hole punching for metadata page